salt/transport/zeromq.py

```diff
@@ -838,15 +844,14 @@ def _publish_daemon(self, log_queue=None):
                     raise exc
 
         except KeyboardInterrupt:
-            # Cleanly close the sockets if we're shutting down
-            if pub_sock.closed is False:
-                pub_sock.setsockopt(zmq.LINGER, 1)
-                pub_sock.close()
-            if pull_sock.closed is False:
-                pull_sock.setsockopt(zmq.LINGER, 1)
-                pull_sock.close()
-            if context.closed is False:
-                context.term()
+            log.trace('Publish daemon caught Keyboard interupt, tearing down')
+        # Cleanly close the sockets if we're shutting down
+        if pub_sock.closed is False:
+            pub_sock.close()
```

Comment:
Hmm... Why not reuse [function you've just created](https://github.com/saltstack/salt/pull/50463/files#diff-7e93fb0d8129a1b378ae1a10559936cbR901)? I would rather have any socket variable closing function that takes a socket variable and closes it with all the precautions and logging, instead of always checking that.

To me it is too much `if/else` and self-repeating, rather then can be just one function reused to all the sockets.

Reply:
This is not the same socket. The socket here is being created in the daemon process. The `pub_sock` you are talking about here is sending things to the minions. I would have named it something different but not trying to change to much as a part of a bug fix PR.